### PR TITLE
OPS-2307 travis-ci and Dockerhub integration

### PR DESCRIPTION
**Pull request by cytopia**
# Travis-CI and Dockerhub integration

Multiple Ansible versions: 2.3, 2.4, 2.5 and latest

## How to build locally
```bash
make build-all
```

## How to test locally
```bash
make test-all
```

## Help?
```bash
make help
```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+###
+### Build Docker images
+###
+before_script:
+  - make update-base
+  - make build-${TAG}
+
+  # Test if it exists
+  - if ! docker images | grep "${image}" | grep "${TAG}"; then
```
Comment (cytopia) [inline .travis.yml]: Travis seems to break on all numeric `TAG` values (`23`, `24` and `25`). Only the one which has `latest` as its value seems to work.
@danvaida Can u provide some help here.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+### Push Docker images
+###
+script:
+  - if [ "${TRAVIS_PULL_REQUEST}" == "false" ]; then
+      echo "${DOCKER_PASSWORD}" | docker login --username "${DOCKER_USERNAME}" --password-stdin &&
+      if [ "${TRAVIS_BRANCH}" == "master" ]; then
+        docker push "${image}:${TAG}";
+      elif [ -n "${TRAVIS_TAG}" ]; then
+        docker tag "${image}:${TAG}" "${image}:${TAG}-${TRAVIS_TAG}";
+        docker push "${image}:${TAG}-${TRtAVIS_TAG}";
```
Comment (danvaida) [inline .travis.yml]: TRtAVIS?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,30 @@
+FROM debian:stretch-slim
```
Comment (danvaida) [inline Dockerfile:1]: will we support Ubuntu in the future?

File: .travis.yml (hunk 2 above)
Comment (cytopia) [reply .travis.yml]: Good catch.

File: Dockerfile (hunk 3 above)
Comment (snovikov) [reply Dockerfile:1]: No

Comment (danvaida) [reply Dockerfile:1]: alright. perhaps I misunderstood what were these images meant for :) AFAIK, at least the shop stack runs on Ubuntu 16.04.

**Review by snovikov — APPROVED** [review]
Nice!

File: Dockerfile (hunk 3 above)
Comment (cytopia) [reply Dockerfile:1]: Is there any reason to have an additional Docker image which does the same, but is just based on a different OS?

Comment (cytopia) [reply Dockerfile:1]: These images are just supposed to use them for running Ansible itself. Either locally or for automated tests

Comment (danvaida) [reply Dockerfile:1]: Ah, I see. That's where the confusion came from. If you just want to use it as a runner, then it's fine, of course.
My use case is to use them to also test Ansible roles on different distributions and their versions.

**Review by danvaida — APPROVED** [review]

File: Dockerfile (hunk 3 above)
Comment (cytopia) [reply Dockerfile:1]: Also a valid use-case. So I'd suggest we merge it now and simply add another Dockerfile as soon as we need it.

**Pull request merged**